backend/storage/discussion-board.sql

```diff
@@ -0,0 +1,108 @@
+CREATE TYPE position_types AS ENUM ('Student', 'TA', 'Instructor');
+
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY,
+  first_name varchar(50),
+  last_name varchar(50),
+  preferred_name varchar(50),
+  email varchar(320) UNIQUE NOT NULL
+);
+
+CREATE TABLE semesters (
+  id SERIAL PRIMARY KEY,
+  semester_season varchar(100) NOT NULL,
+  semester_year int NOT NULL
+);
+
+CREATE TABLE institutions (
+  id SERIAL PRIMARY KEY,
+  name varchar(100) NOT NULL,
+  location varchar(100)
+);
+
+CREATE TABLE administrators (
+  user_id int,
+  institution_id int,
+  PRIMARY KEY (user_id, institution_id)
+);
+
+CREATE TABLE enrolment (
+  user_id int NOT NULL,
+  course_id int NOT NULL,
+  position position_types NOT NULL,
+  PRIMARY KEY (user_id, course_id)
+);
+
+CREATE TABLE courses (
+  id SERIAL PRIMARY KEY,
+  course_name varchar(100),
+  course_code varchar(50) NOT NULL,
+  semester_id int,
+  is_archived boolean
+);
+
+CREATE TABLE categories (
+  id SERIAL PRIMARY KEY,
+  category_name varchar(100) NOT NULL,
+  course int
+);
+
+CREATE TABLE post_categories (
+  category_id int,
+  post_id int,
+  PRIMARY KEY (category_id, post_id)
+);
+
+CREATE TABLE posts (
+  id SERIAL PRIMARY KEY,
+  author int,
+  parent_id int,
+  num_children int DEFAULT 0,
+  title varchar(200) NOT NULL,
+  body varchar,
+  image int,
+  upvotes int DEFAULT 0,
+  view_count int DEFAULT 0,
+  created_at timestamp with time zone DEFAULT (now()),
+  updated_at timestamp with time zone DEFAULT (now())
+);
```

Comment:
Are we allowing nested comments? What would a query look like for all posts, it might get very difficult to reason about queries on this

Reply:
For the second question, now each post references its course, since a post shouldn't be about more than one course (because posts belong to categories which are unique to courses). So it should be straightforward to find all posts under a course, even without filtering by category.

For the first question, a separate "replies" table makes sense since top-level posts have some extra fields... but then the parent_id could be either a post or a reply... or maybe just limit the nesting of comments on the front end?

Reply:
If we want to try and find a happy medium between SO and piazza, we might want to differentiate between replies to a post, and the comments below that reply. The comments can be much simpler, just like SO comments, what do you think about this idea?

Reply:
Yup, makes sense to differentiate between posts/replies and comments! Added a "comments" table

Reply:
Since posts and replies (not comments) have similar fields except title and view_count, they can be in the same table right? That way, awards can be given to both posts and replies.